src/main/kotlin/de/bund/digitalservice/useid/filter/SimpleAuthenticationFilter.kt

```diff
@@ -0,0 +1,26 @@
+package de.bund.digitalservice.useid.filter
+
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.web.server.ResponseStatusException
+import org.springframework.web.server.ServerWebExchange
+import org.springframework.web.server.WebFilter
+import org.springframework.web.server.WebFilterChain
+import reactor.core.publisher.Mono
+
+class SimpleAuthenticationFilter : WebFilter {
+    override fun filter(serverWebExchange: ServerWebExchange, webFilterchain: WebFilterChain): Mono<Void> {
+        val authorizationHeader: String? = serverWebExchange.request.headers.getFirst(HttpHeaders.AUTHORIZATION)
+
+        // TODO: Is there another way writing this code below more reactive?
+        return when {
+            !isValidHeader(authorizationHeader) -> Mono.error(ResponseStatusException(HttpStatus.UNAUTHORIZED, "Missing: Invalid header"))
```

Comment:
The error message sounds a bit weird (Missing or invalid?).

Reply:
true it should have been `error: invalid header`